pkg/curatedpackages/reader.go

```diff
@@ -28,16 +30,25 @@ type ManifestReader interface {
 
 type PackageReader struct {
 	ManifestReader
+	AllImages     bool
+	storageClient registry.StorageClient
 }
 
+var _ ManifestReader = (*PackageReader)(nil)
+
 func NewPackageReader(mr ManifestReader) *PackageReader {
 	return &PackageReader{
 		ManifestReader: mr,
+		AllImages:      true,
 	}
 }
 
 func (r *PackageReader) ReadImagesFromBundles(ctx context.Context, b *releasev1.Bundles) ([]releasev1.Image, error) {
-	images, err := r.ManifestReader.ReadImagesFromBundles(ctx, b)
+	var err error
+	var images []releasev1.Image
+	if r.AllImages {
```

Comment:
It's unclear what the difference is between ReadImagesFromBundles and fetchImagesFromBundles.

I'm guessing the former are for curated packages, while the latter are for EKS-A core images?

If so, perhaps some renaming could make it clearer what's going on:

```
r.AllImages => r.IncludePackagesImages
r.ManifestReader.ReadImagesFromBundles => ReadPackagesImagesFromBundles
```

Reply:
I agree this code is odd, but I don't want to refactor it.